qiskit/pulse/library/symbolic_pulses.py

```diff
@@ -0,0 +1,849 @@
+
+        Raises:
+            PulseError: When parameters are not bound.
+        """
+        if self.is_parameterized():
+            raise PulseError(
+                f"Unassigned parameter exists: {self.parameters}. All parameters should be assigned."
+            )
+
+        times = np.arange(0, self.duration) + 1 / 2
+        args = (times, *self.param_values)
+        waveform = self.envelope(*args)
+
+        return Waveform(samples=waveform, name=self.name)
+
+    def validate_parameters(self) -> None:
+        """
+        Validate parameters.
+
+        Raises:
+            PulseError: If the parameters passed are not valid.
+        """
+        args = (self.limit_amplitude, *self.param_values)
+
+        for i, constraint in enumerate(self.constraints):
+            if isinstance(constraint, list):
+                # Cannot use sympy.ITE because it doesn't support lazy evaluation.
+                # See https://github.com/sympy/sympy/issues/23295
+                eval_res = _evaluate_ite(constraint, *args)
+            else:
+                eval_res = bool(constraint(*args))
+            if not eval_res:
+                dict_repr = ", ".join(f"{p} = {v}" for p, v in self.parameters.items())
+                const = str(self._define_constraints()[i])
+                raise PulseError(
+                    f"Assigned parameters {dict_repr} violate following constraint: {const}."
+                )
+
+    def is_parameterized(self) -> bool:
+        """Return True iff the instruction is parameterized."""
+        return any(isinstance(val, ParameterExpression) for val in self.param_values)
+
+    @property
+    def parameters(self) -> Dict[str, Any]:
+        return dict(zip(self.PARAM_DEF, self.param_values))
+
+    def __eq__(self, other: Pulse) -> bool:
+        return super().__eq__(other) and self.parameters == other.parameters
+
+    def __hash__(self) -> int:
+        return hash(tuple(self.parameters[k] for k in sorted(self.parameters)))
+
+    def __repr__(self) -> str:
+        params_str = ", ".join(f"{p}={v}" for p, v in zip(self.PARAM_DEF, self.param_values))
+
+        return "{}({}{})".format(
+            self.__class__.__name__,
+            params_str,
+            f", name='{self.name}'" if self.name is not None else "",
+        )
+
+
+class Gaussian(SymbolicPulse):
+    r"""A lifted and truncated pulse envelope shaped according to the Gaussian function whose
+    mean is centered at the center of the pulse (duration / 2):
+
+    .. math::
+
+        f'(x) &= \exp\Bigl( -\frac12 \frac{{(x - \text{duration}/2)}^2}{\text{sigma}^2} \Bigr)\\
+        f(x) &= \text{amp} \times \frac{f'(x) - f'(-1)}{1-f'(-1)}, \quad 0 \le x < \text{duration}
+
+    where :math:`f'(x)` is the gaussian waveform without lifting or amplitude scaling.
+
+    This pulse would be more accurately named as ``LiftedGaussian``, however, for historical
+    and practical DSP reasons it has the name ``Gaussian``.
```

Comment:
what is DSP? Please spell it out.

Reply:
I guess digital signal processor. I just copied this function from discrete module. To be honest I don't know that is the DSP reason. I'll remove this sentence since likely this explains some IBM convention.

Reply:
I think the point that sentence is making is that for a physical DSP it is much more efficient to limit pulses to finite duration instead of extending them to infinity. Wherever you cut off a Gaussian, there will be a step discontinuity for the samples before and after the pulse. To avoid this, the pulse is lifted (really "lowered" would be more accurate) so that the edges step smoothly to zero.